### PR TITLE
lib: fix some type mismatches and remove unneeded typecasts

### PR DESCRIPTION
**Pull request by jay**
Many of these castings are unneeded if we change the variables to work better with each other.

Ref: https://github.com/curl/curl/pull/9823

Closes #xxxx

---

This is a subset of the changes submitted by @AtariDreams in #9823.

### CODE REVIEW

File: lib/ftp.c [hunk 1]
```diff
@@ -4167,7 +4167,7 @@ CURLcode ftp_parse_url_path(struct Curl_easy *data)
         /* get path before last slash, except for / */
         size_t dirlen = slashPos - rawPath;
         if(dirlen == 0)
-            dirlen++;
+            dirlen = 1;
```
Comment (bagder) [inline lib/ftp.c:4170]: maybe fix the indent level while at it!

File: lib/noproxy.c [hunk 2]
```diff
@@ -222,7 +222,12 @@ bool Curl_check_noproxy(const char *name, const char *no_proxy)
           slash = strchr(check, '/');
           /* if the slash is part of this token, use it */
           if(slash) {
-            bits = atoi(slash + 1);
+            unsigned long longbits = strtoul(slash + 1, NULL, 10);
+            if(longbits > UINT_MAX)
+              /* this cannot match */
+              break;
```
Comment (bagder) [inline lib/noproxy.c:228]: The cidr match functions already return error if the bits number is too big.

File: lib/http.c [hunk 3]
```diff
@@ -650,7 +650,7 @@ CURLcode Curl_http_auth_act(struct Curl_easy *data)
   if(!data->set.str[STRING_BEARER])
     authmask &= (unsigned long)~CURLAUTH_BEARER;
 
-  if(100 <= data->req.httpcode && 199 >= data->req.httpcode)
+  if(100 <= data->req.httpcode && data->req.httpcode <= 199)
```
Comment (MarcelRaad) [inline lib/http.c:653]: Not sure what this is supposed to fix. The rest looks good to me!

**Review by MarcelRaad — APPROVED** [review]

File: lib/http.c (hunk 3 above)
Comment (jay) [reply lib/http.c:653]: This one isn't supposed to fix anything it's just a style change.

**Pull request closed without merging**